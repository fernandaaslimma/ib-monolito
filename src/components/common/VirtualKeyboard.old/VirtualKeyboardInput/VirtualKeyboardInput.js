import React, { Component } from "react";
import throttle from "lodash/throttle";

let closeTimeout;

class VirtualKeyboardInput extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isDetached: false
    };
  }

  componentDidMount() {
    this.checkDetachment();

    this.listener = throttle(() => {
      this.checkDetachment();
    }, 500);

    window.addEventListener("resize", this.listener);
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.listener);
  }

  checkDetachment() {
    const isDetached =
      this.props.detachBelow && window.innerWidth < this.props.detachBelow;

    this.setState({ isDetached });
  }

  render() {
    const { render, setActiveInput, cleanActiveInput } = this.props;

    const self = this;
    const RenderedComponent = render();

    let innerRef;
    const NewCompoent = React.cloneElement(RenderedComponent, {
      innerRef(el) {
        innerRef = el;
      },
      readOnly: self.state.isDetached ? null : "readonly",
      onKeyDown(e) {
        if (self.state.isDetached) {
          RenderedComponent.onKeyDown && RenderedComponent.onKeyDown(e);
          return;
        }

        e.preventDefault();
      },
      onBlur(e) {
        if (self.state.isDetached) {
          RenderedComponent.onBlur && RenderedComponent.onBlur(e);
          return;
        }
        clearTimeout && clearTimeout(closeTimeout);
        closeTimeout = setTimeout(() => {
          setActiveInput && cleanActiveInput(innerRef);
        }, 200);
      },
      onFocus(e) {
        if (self.state.isDetached) {
          RenderedComponent.onFocus && RenderedComponent.onFocus(e);
          return;
        }
        clearTimeout && clearTimeout(closeTimeout);
        setActiveInput && setActiveInput(innerRef, NewCompoent);
      }
    });

    return NewCompoent;
  }
}

VirtualKeyboardInput.cleanTimeout = function() {
  clearTimeout && clearTimeout(closeTimeout);
};

export default VirtualKeyboardInput;
