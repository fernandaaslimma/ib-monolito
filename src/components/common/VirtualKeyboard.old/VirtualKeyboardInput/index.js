import VirtualKeyboardInput from "./VirtualKeyboardInput";

export default VirtualKeyboardInput;
