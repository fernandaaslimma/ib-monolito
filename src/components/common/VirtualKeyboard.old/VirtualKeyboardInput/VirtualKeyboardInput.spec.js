import React from "react";
import { shallow } from "enzyme";

import VirtualKeyboardInput from "./VirtualKeyboardInput";

describe("VirtualKeyboardInput component", () => {
  it("Should match snapshot", () => {
    const virtualKeyboardInput = shallow(
      <VirtualKeyboardInput render={() => <input />} />
    );

    expect(virtualKeyboardInput).toMatchSnapshot();
  });

  it("Should render a wrapped component", () => {
    const virtualKeyboardInput = shallow(
      <VirtualKeyboardInput render={() => <input />} />
    );

    expect(virtualKeyboardInput.html()).toEqual('<input readonly=""/>');
  });

  it("Should render a wrapped component with custom onkeyDown prop", () => {
    const virtualKeyboardInput = shallow(
      <VirtualKeyboardInput render={() => <input />} />
    );

    expect(
      typeof virtualKeyboardInput.props().onKeyDown === "function"
    ).toBeTruthy();
  });

  it("Should call custom onkeyDown prop with prevent default", () => {
    const onKeyDownEvent = {
      preventDefault: jest.fn()
    };

    const virtualKeyboardInput = shallow(
      <VirtualKeyboardInput render={() => <input />} />
    );

    virtualKeyboardInput.simulate("keydown", onKeyDownEvent);
    expect(onKeyDownEvent.preventDefault).toHaveBeenCalled();
  });

  it("Should not call custom onkeyDown prop with prevent default if window.innerWidth is lower than detachBelow prop", () => {
    global.window = global.window || {};
    global.window.innerWidth = 400;

    const onKeyDownEvent = {
      preventDefault: jest.fn()
    };

    const virtualKeyboardInput = shallow(
      <VirtualKeyboardInput detachBelow={500} render={() => <input />} />
    );

    virtualKeyboardInput.simulate("keydown", onKeyDownEvent);
    expect(onKeyDownEvent.preventDefault).not.toHaveBeenCalled();
  });

  it("Should call custom onkeyDown prop with prevent default if window.innerWidth is greater than detachBelow prop", () => {
    global.window = global.window || {};
    global.window.innerWidth = 700;

    const onKeyDownEvent = {
      preventDefault: jest.fn()
    };

    const virtualKeyboardInput = shallow(
      <VirtualKeyboardInput detachBelow={500} render={() => <input />} />
    );

    virtualKeyboardInput.simulate("keydown", onKeyDownEvent);
    expect(onKeyDownEvent.preventDefault).toHaveBeenCalled();
  });
});
