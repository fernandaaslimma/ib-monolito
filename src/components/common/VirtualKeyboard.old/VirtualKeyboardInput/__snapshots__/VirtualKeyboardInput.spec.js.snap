// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VirtualKeyboardInput component Should match snapshot 1`] = `
<input
  innerRef={[Function]}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  readOnly="readonly"
/>
`;
